### PR TITLE
chore: update versioning and changelog for release version 2.3.3

### PR DESCRIPTION
**Pull request by aaronchung-bitquill**
### Summary
chore: update versioning and changelog for release version 2.3.3

### Description
chore: update versioning and changelog for release version 2.3.3

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md [hunk 1]
```diff
@@ -196,7 +196,7 @@ If there is an unreleased feature you would like to try, it may be available in
   <dependency>
     <groupId>software.amazon.jdbc</groupId>
     <artifactId>aws-advanced-jdbc-wrapper</artifactId>
-    <version>2.3.2-SNAPSHOT</version>
+    <version>2.3.3-SNAPSHOT</version>
```
Comment (crystall-bitquill) [inline docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:199]: The snapshot version should be ahead of the current, so 2.3.4 in this case I believe

File: gradle.properties [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
-aws-advanced-jdbc-wrapper.version.major=2
+aws-advanced-jdbc-wrapper.version.major=3
```
Comment (crystall-bitquill) [inline gradle.properties:15]: This changes the major version, it should be for the subminor line

File: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md (hunk 1 above)
Comment (aaronchung-bitquill) [reply docs/using-the-jdbc-driver/UsingTheJdbcDriver.md:199]: Good catch!

Comment (aaronchung-bitquill) [reply docs/using-the-jdbc-driver/UsingTheJdbcDriver.md]: Fixed

File: gradle.properties (hunk 2 above)
Comment (aaronchung-bitquill) [reply gradle.properties]: Fixed

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**